doc/developing_oemof.rst

```diff
@@ -51,31 +51,51 @@ Newly added required packages (via PyPi) are installed by performing a manual up
 
   pip3 install --upgrade -e /path/to/the/repository
   
-Documentation
--------------
+Contribute to the documentation
+-------------------------------
 
 See the developer version of the documentation of the dev branch at
 `readthedocs.org <http://oemof.readthedocs.org/en/latest/>`_.
 
+Contribute to new components
+----------------------------
+                                                                                                                                                       
+You can develop a new component according to your needs. Therefore you can use the module oemof.solph.customs which collects custom components created by users and lowers the entry barrier for contributing.                
+                                 
+Your code should fit to the :ref:`style_guidlines_label`, but not necessarily to the :ref:`naming_conventions_label`. Also compatiblity to the results-
+API must not be guaranteed. Further you do not need to test your components or adapt the documentation. These steps are all necessary once your custom
+component becomes a constant part of oemof (oemof.solph.components) and are described here: :ref:`_coding_requirements_label`. But in the first step have a look at existing custom components created by other users in oemof.solph.custom and easily create your own if you need.     
 
 Collaboration with pull requests
 --------------------------------
 
-To collaborate use the pull request functionality of github.
+To collaborate use the pull request functionality of github as described here: https://guides.github.com/activities/hello-world/
+
+How to create a pullrequest
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* Fork the oemof repository to your own github account.
+* Change, add or remove code.
+* Commit your changes.
+* Create a pull request and describe what you will do and why.
+* Choose reviewers and assignees.
+* Optionally assign a label and milestone.
+* Wait for approval.
 
-How to create a pull request
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-* Fork the oemof repository
-* Create a pull request and describe what you will do and why
-* Change, add or remove code
-* Read the :ref:`style_guidlines_label` and follow them
-* Add new tests according to what you have done
+.. _coding_requirements_label:  
+
+Generally the following steps are required when changing, adding or removing code
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+* Read the :ref:`style_guidlines_label` and :ref:`naming_conventions_label` and follow them
+* Add new :ref:`tests_label` according to what you have done
 * Add/change the documentation (new feature, API changes ...)
 * Add a whatsnew entry and your name to Contributors
-* Check if all tests still work including the example tests
+* Check if all tests still work including the example files in `oemof_examples <https://github.com/oemof/oemof_examples/>`_.
```

Comment:
As we decided to move the examples to the example repository we decided that they do not have a test purpose any more. If you need a test you have to write a test, if you need an example you write an example.

Nosetests are sufficient.

```
nosetests -w path/to/oemof --with-doctest
```